### DISCUSSION TITLE
How do we publish from a test environment to a production environment

### DISCUSSION THREAD

**Discussion by lenboo** [apache/dolphinscheduler#6724] [General]
How do we publish from a test environment to a production environment？

**Comment by Tandoy** [comment]
This is also our pain point. If we need to publish to production, we need to export the workflow and then import JSON. Does the community consider developing new features that provide environmental isolation and release to production like Alibaba's dataworks.

**Reply by ruanwenjun** [reply]
A simple way is just providing a button to release to production, once click the button, it will call the OpenAPI to create workflow in production or create workflow in xx environment. This may require the test environment network can connect to production.
